server/images.go

```diff
@@ -56,6 +63,21 @@ func (m *Model) Prompt(request api.GenerateRequest) (string, error) {
 	vars.Prompt = request.Prompt
 	vars.Context = request.Context
 
+	if len(m.Embeddings) > 0 {
+		promptEmbed, err := loaded.llm.Embedding(request.Prompt)
+		if err != nil {
+			return "", fmt.Errorf("failed to get embedding for prompt: %v", err)
+		}
+		// TODO: set embed_top from specified parameters in modelfile
+		embed_top := 3
+		embed := vector.TopK(embed_top, mat.NewVecDense(len(promptEmbed), promptEmbed), loaded.Embeddings)
+		toEmbed := ""
+		for _, e := range embed {
+			toEmbed = fmt.Sprintf("%s %s", toEmbed, e.Embedding.Data)
+		}
+		vars.Embed = toEmbed
+	}
```

Comment:
I think the bulk of the code here should be moved to the caller. There's no guarantee `loaded.llm` is non-nil. It'll be better if the caller calculates `toEmbed` and passes it in to `Prompt` as a parameter.

Reply:
good call, moved it